src/parser.cpp

```diff
@@ -391,7 +391,16 @@ void Parser::ParserImpl::loadModel(const ModelPtr &model, const std::string &inp
         loadConnection(model, connectionNode);
     }
 
-    model->linkUnits();
+    auto logger = Logger::create();
+    for (size_t index = 0; index < model->componentCount(); ++index) {
+        auto c = model->component(index);
+        traverseComponentTreeLinkingUnits(c, logger);
```

Comment:
Is there a cleaner way to pass the private logger in the parser into an external function?

Reply:
Pass `Parser` shared_from_this() instead of logger.